src/components/SchemaRow.tsx

```diff
@@ -1,192 +1,67 @@
-              boundary="window"
-              interactionKind="hover"
-              className="ml-2 flex-1 truncate flex items-baseline"
-              target={<div className="text-darken-7 dark:text-lighten-7 w-full truncate">{description}</div>}
-              targetClassName="text-darken-7 dark:text-lighten-6 w-full truncate"
-              content={
-                <div className="p-5" style={{ maxHeight: 500, maxWidth: 400 }}>
-                  <MarkdownViewer markdown={description} />
-                </div>
-              }
-            />
-          )}
+          <Property node={schemaNode} onGoToRef={onGoToRef} />
+          {description && <Description value={description} />}
         </div>
 
-        {validationCount ? (
-          <Popover
-            boundary="window"
-            interactionKind="hover"
-            content={
-              <div className="p-5" style={{ maxHeight: 500, maxWidth: 400 }}>
-                {map(Object.keys(nodeValidations), (key, index) => {
-                  const validation = nodeValidations[key];
-
-                  let elem = null;
-                  if (Array.isArray(validation)) {
-                    elem = validation.map((v, i) => (
-                      <div key={i} className="mt-1 mr-1 flex items-center">
-                        <div className="px-1 bg-gray-2 dark:bg-gray-8 font-bold text-sm rounded">{String(v)}</div>
-                        {i < validation.length - 1 ? <div>,</div> : null}
-                      </div>
-                    ));
-                  } else if (typeof validation === 'object') {
-                    elem = (
-                      <div className="m-1 px-1 bg-gray-2 dark:bg-gray-8 font-bold text-sm rounded" key={index}>
-                        {'{...}'}
-                      </div>
-                    );
-                  } else {
-                    elem = (
-                      <div className="m-1 px-1 bg-gray-2 dark:bg-gray-8 font-bold text-sm rounded" key={index}>
-                        {JSON.stringify(validation)}
-                      </div>
-                    );
-                  }
-
-                  return (
-                    <div key={index} className="py-1 flex items-baseline">
-                      <div className="font-medium pr-2 w-24">{key}:</div>
-                      <div className="flex-1 flex flex-wrap text-center">{elem}</div>
-                    </div>
-                  );
-                })}
-              </div>
-            }
-            target={requiredElem}
-          />
-        ) : (
-          requiredElem
-        )}
-        {rowRendererRight && <div className="ml-2">{rowRendererRight(node)}</div>}
+        <Validations
+          required={!!schemaNode.required}
+          validations={{
+            ...('annotations' in schemaNode &&
+              schemaNode.annotations.default && { default: schemaNode.annotations.default }),
+            ...('validations' in schemaNode && schemaNode.validations),
```

Comment:
@P0lip it might be just me but I find it a bit intimidating to read. Would you mind wrapping this with a small helper function? E.g. `getAnnotations()` + `getValidations()`?

Reply:
Sounds good to me.